agent/whatweb_agent.py

```diff
@@ -81,38 +81,97 @@ def process(self, message: msg.Message) -> None:
             message:  The message to process from ostorlab runtime.
         """
         logger.info('processing message of selector : %s', message.selector)
-        target = self._prepare_target(message)
-        is_target_already_processed = self._is_target_already_processed(message)
-        if is_target_already_processed is False:
-            return
-        else:
-            with tempfile.NamedTemporaryFile() as fp:
-                self._start_scan(target.domain, fp.name)
-                self._parse_emit_result(target.domain, fp, int(target.port), target.schema)
+        targets = self._prepare_targets(message)
+        for target in targets:
+            is_target_already_processed = self._is_target_already_processed(
```

Comment:
no need to use the variable is_target_already_processed, you can do the if check directly on the function result